### ISSUE TITLE
Searching stories by hashtag/usertag issue

### ISSUE THREAD

**Issue by AndManowar** [mgp25/Instagram-API#2391]
Hello, is it possible to search stories by hashtag or usertag?
I know about *$client->hashtag->getStory()* method, but can i paginate stories somehow? I use *markMediaSeen()* in hashtag  but Im getting old stories too, can you explain me?

Code:
    
      $maxId = null;
        do {
            $feed = $this->client->hashtag->getFeed('cat', Signatures::generateUUID(), $maxId);
            if (is_null($feed->getStory())) {
                echo 'No available stories' . PHP_EOL;
                return;
            }

            foreach ($feed->getStory()->getItems() as $story) {

                if ($story->getIsSeen()) {
                    echo 'story is seen ' . $story->getUser()->getUsername();
                    continue;
                }

                if ($this->isUserInBlackList($story->getUser()->getUsername())) {
                    echo 'User is in black list' . PHP_EOL;
                    continue;
                }

                if ($this->isEntityAlreadyParsed($story->getId(), $story->getUser()->getPk(), ParsedCompetitor::ACTION_STORIES, $story->getUser()->getUsername())) {
                    echo 'Story was already parsed ' . $story->getUser()->getUsername() . PHP_EOL;
                    continue;
                }
                $competitor = $this->getCompetitorModel($story->getUser()->getUsername(), $story->getUser()->getPk());
                $this->addScoreForCompetition($competitor, 'story_value', true);
                $this->setParsedAction($competitor, $story->getId(), $story->getUser()->getPk(), ParsedCompetitor::ACTION_STORIES);
            }

            $maxId = $feed->getNextMaxId();
            $this->client->hashtag->markStoryMediaSeen($feed, $feed->getStory()->getItems());
        } while (count($feed->getStory()->getItems()) > 0 && $maxId !== null);

**Comment by AndManowar** [comment]
When this is not popular hashtag - everything is OK, but when this is popular like cat or dog - Im getting "Story was already parsed" message. The story id is not unique?

**Comment by Ant-V** [comment]
This is just from the top go my head, but did you try setting the signature outside of the loop so you can keep it the same across all loops?

**Comment by AndManowar** [comment]
> This is just from the top go my head, but did you try setting the signature outside of the loop so you can keep it the same across all loops?

No, it doesnt help

**Comment by mgp25** [comment]
Seems a normal behavior, check Instagram apps, you will see the same stories even if you have already seen them

Regards

**Comment by romshiri** [comment]
> Seems a normal behavior, check Instagram apps, you will see the same stories even if you have already seen them
> 
> Regards

When checking instagram app - I see that every "seen" video (manually) is usually switched and the next time I see it (whether I fetch it programmatically or refreshing the story's tray) is after a few iterations. 

When I mark story as "seen" using the API, when I fetch the stories again, I still see it when fetching it (programmatically or refreshing manually) which makes it impossible to iterate through stories result.

Ways to reproduce ("stories do iterate" case):
1. Go to instagram app/web 
2. Search for hashtag story
3. Fetch the stories of this hashtag  via the API -> see results
4. Watch stories manually on IG app/web.
5. Fetch the stories again via the API -> results without the stories you had watched

Ways to reproduce ("stories do not iterate" case):
1. Fetch the stories of this hashtag via the API -> see results
2. Mark several stories as "seen" using the API.
3. Fetch the stories again via the API -> same results, no iteration

I wonder what happens differently on web/app that makes the results to exclude the stories I've just watched manually, but to still include the stories I marked as "seen" using the API. 

Does anyone have any insights on this?
